debezium-core/src/main/java/io/debezium/util/TimeBasedRateLimiter.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.util;
+
+import java.time.Duration;
+
+import io.debezium.annotation.NotThreadSafe;
+
+/**
+ * Maintains information about rate limiting for an action; for instance, can be used for limiting the number of certain
+ * log messages to max. 1 per 10 seconds.
+ * <p/>
+ * Each instance of this class is meant to correspond to a single action that is to be rate limited. The action is not
+ * referenced in this class - the limiter expects that its {@link #allows()} method is called whenever
+ * the "action under limit" is to be called and the client wants to determine if the call would exceed the limit
+ * set by the limiter.
+ * <p/>
+ * Example:
+ * <pre>
+ * {@code
+ *   limiter = new TimeBasedRateLimiter(Duration.ofSeconds(5)) // interval length 5 seconds
+ *   Time: +0s limiter.allows() == true // interval start
+ *   Time: +1s limiter.allows() == false // within 5 sec interval, the action should be limited
+ *   Time: +2s limiter.allows() == false // within 5 sec interval, the action should be limited
+ *   ...
+ *   Time: +5s limiter.allows() == true // out of 5 sec interval, the action should NOT be limited, next interval starts
+ *   Time: +6s limiter.allows() == false // within 5 sec interval, the action should be limited
+ *   ...
+ *   Time: +10s limiter.allows() == true // out of 5 sec interval, the action should NOT be limited, next interval starts
+ *   ...
+ * }
+ * </pre>
+ */
+@NotThreadSafe
```

Comment:
The class is not designed to be thread safe - if the `AbstractMessageDecoder` needs thread safety of this class, please let me know.